lib/Doctrine/Common/DataFixtures/Loader.php

```diff
@@ -181,7 +183,7 @@ public function addFixture(FixtureInterface $fixture)
     /**
      * Returns the array of data fixtures to execute.
      *
-     * @psalm-return array<class-string<OrderedFixtureInterface>|int, OrderedFixtureInterface>
+     * @psalm-return array<class-string<OrderedFixtureInterface&FixtureInterface>|int, OrderedFixtureInterface&FixtureInterface>
```

Comment:
```suggestion
     * @psalm-return array<class-string<OrderedFixtureInterface|FixtureInterface>|int, OrderedFixtureInterface&FixtureInterface>
```

Is it not `|` also this still wont fix the phpstan error but that should be done in major release right as that would be a breaking change?

Reply:
> Is it not `|`

No? Why on earth would it? It's not an `OrderedFixtureInterface` OR a `FixtureInterface`, it is both, right?


>  this still wont fix the phpstan error

why not?

Reply:
> > Is it not `|`
> 
> No? Why on earth would it? It's not an `OrderedFixtureInterface` OR a `FixtureInterface`, it is both, right?

I think your right because `OrderedFixtureInterface` and `FixtureInterface` would both be implemented
> 
> > this still wont fix the phpstan error
> 
> why not? have you tried it?

No but i see you update the phpstan ignore error so i assumed its not fixed, but dont know what we should change without breaking to fix it

Reply:
> > have you tried it?

> No 

Well what are you waiting for?

Reply:
> > > have you tried it?
> 
> > No
> 
> Well what are you waiting for?

I confirmed it works for my case, but i still have the question why we need to keep these ignored https://github.com/doctrine/data-fixtures/pull/416/files#diff-995edee38ad4f8387e58ebd52c31bcc04c56cc2448d331b1cf5e0b35c57b9efaR29?

Reply:
Fair question, let me take a closer look.

Reply:
:thinking: I know think what has been done in https://github.com/doctrine/data-fixtures/pull/389/files#diff-7d95dacf620e8b57cfdb926d6f6cc52dbf52687cf3095881544bf9c6e3abc4d8 was not only imprecise, but also plain wrong: `getFixtures` may return fixtures that are not ordered fixtures! The property is called `orderedFixtures` can contain fixtured ordered because the fixtures implement `OrderedFixtureInterface`, or because they implement `DependentFixtureInterface`, and if they don't, the order will be the order in which they were added. Thanks for making me take a closer look, I'll change this PR!